examples/column_view_datagrid/grid_cell/imp.rs

```diff
@@ -0,0 +1,31 @@
+use gtk::glib;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+
+use gtk::CompositeTemplate;
+
+#[derive(Debug, Default, CompositeTemplate)]
+#[template(file = "grid_cell.ui")]
+pub struct GridCell {
+    #[template_child]
+    pub name: TemplateChild<gtk::Label>,
+}
+
+#[glib::object_subclass]
+impl ObjectSubclass for GridCell {
+    const NAME: &'static str = "GridCell";
+    type Type = super::GridCell;
+    type ParentType = gtk::Box;
```

Comment:
I would make this a gtk::Widget and set the layout manager using `klass.set_layout_manager::<gtk::BinLayout>();` instead. Note you will have to unparent the child widget manually on dispose.

Reply:
i think I implemented this now, I looked at other examples in the codebase that have a dispose on ObjectImpl